plugins/modules/ec2_instance.py

```diff
@@ -1699,6 +1751,48 @@ def handle_existing(existing_matches, state):
     return result
 
 
+def enforce_count(existing_matches, module):
+    exact_count = module.params.get('exact_count')
+
+    try:
+        current_count = len(existing_matches)
+        if current_count == exact_count:
+            module.exit_json(
+                changed=False,
+                msg='{0} instances already running, nothing to do.'.format(exact_count)
+            )
+
+        elif current_count < exact_count:
+            to_launch = exact_count - current_count
+            module.params['to_launch'] = to_launch
+            # launch instances
+            try:
+                ensure_present(existing_matches=existing_matches, desired_module_state='present')
+            except botocore.exceptions.ClientError as e:
```

Comment:
These days we often try to catch `BotoCoreError` too.  This is most likely to appear if there's a botocore version mismatch and a feature that's being relied on isn't available in the installed version.

At the minute we only warn on old versions that aren't supported by the collection.  We error when the collection supports the version, but the specific feature needs a version newer than our 'supported' version.